### ISSUE TITLE
Git Process: fatal: not a git repository

### ISSUE THREAD

**Issue by riasvdv** [statamic/cms#3262]
## Bug Description

When running the git integration, I'm getting a lot of "Git Process: fatal: not a git repository" exceptions logged.

https://flareapp.io/share/w5BY8AV7#F59

The integration is working however, changes are being pushed correctly.

Jobs are pushed to a redis queue:
```
STATAMIC_GIT_ENABLED=true
STATAMIC_GIT_AUTOMATIC=true
STATAMIC_GIT_PUSH=true
STATAMIC_GIT_DISPATCH_DELAY=10
```

What might be a cause is that we deploy using atomic deployments. All code runs in `/home/current` which is a symlink to `/home/releases/y-m-d-h-i-s/`

Statamic 3.0.42 Pro
Laravel 8.28.1
PHP 8.0.0

**Comment by jesseleite** [comment]
Weird, I cannot reproduce this. I'm serving through a symlink on valet, but maybe valet is doing something different than your production server 🤔

**Comment by duncanmcclean** [comment]
We're also experiencing this same issue, again when using Envoyer.

**Comment by duncanmcclean** [comment]
As it turns out, we actually don't have a `.git` directory in our Envoyer release folder, which is why we're running into this issue. 

I've submitted a support request to the Laravel team, hopefully there's some way we can retain the `.git` directory.

**Comment by jasonvarga** [comment]
I wonder if @riasvdv is also using Envoyer and seeing the same issue. 

> The integration is working however, changes are being pushed correctly.

Pushed to production? Or pushed from production?

**Comment by riasvdv** [comment]
Nop we use Envoy (https://laravel.com/docs/master/envoy) and that keeps a .git repository intact

**Comment by crnkovic** [comment]
@riasvdv @duncanmcclean Did any of you guys solve this problem?

**Comment by duncanmcclean** [comment]
> @riasvdv @duncanmcclean Did any of you guys solve this problem?

No, we just moved to a separate CMS site, running at `cms.domain.com` which pushes any Git integration changes to the Envoyer sites.

Envoyer doesn't keep the `.git` directory.

**Comment by crnkovic** [comment]
Oh. It's weird -- I do have the .git directory, and yet I still get the same exception. I just use the current, releases and storage directory structure.

**Comment by codedge** [comment]
I can confirm this issue using [Deployer](https://deployer.org/) which also uses the current -> releases/{version} symlink. There is a `.git` folder present but Statamic outputs this error.

And yes, with Valet everythings runs without issues. So there seems to be sth. Valet does different.

**Comment by codedge** [comment]
I was able to track down my error. In `config/statamic/git.php` I had the `public/asset` path configured:

```
'paths' => [
        // ... others
        //public_path('assets'), Symlink to hold media files. Should not be tracked <-- here was the error
    ],
```

But this path is a symlink and should not be tracked. So Statamic and its git implementation was quite right, but the error message which folder is the problematic one is not obvious.

**Comment by jasonvarga** [comment]
Ooh good find. I wonder if @riasvdv has the same issue.

**Comment by riasvdv** [comment]
Our assets are also symlinked, so that could definitely be the case

**Comment by helloiamlukas** [comment]
Had the same problem, removing paths that are symlinked fixed it for me! Thank you @codedge!

**Comment by jesseleite** [comment]
Thanks for the helpful info @codedge @riasvdv @helloiamlukas. With that fix I just pushed, you shouldn't need to remove the symlinked path anymore, _unless_ you want to stop tracking it 👍

**Comment by riasvdv** [comment]
Awesome 🙌
